mongodbatlas/database_users.go

```diff
@@ -48,6 +58,20 @@ type DatabaseUser struct {
 	Username        string  `json:"username,omitempty"`
 }
 
+// GetAuthDB by now just two dbs admin and &external
+func (user *DatabaseUser) GetAuthDB() (name string) {
```

Comment:
[nit] does this has any use in the terraform provider? if not I'd rather not export it

Reply:
Right it's not used there, but if we break the backward compatibility , because the parameter is optional, the plugin will need to figure it out this, calling this function.

Reply:
how are you guys doing with delete? flagging in case you may have the same bug for delete as delete you have to pass the authdb
I'm ok with exporting this if we improve on the doc a little
```
// GetAuthDB determines the authentication database based on the type of user.
// LDAP, X509 and AWSIAM should all use $external.
// SCRAM-SHA should use admin
```
kind of thing

Reply:
okay, checked we cached the id for the delete, and we passed it. just one question in the schema in the terraform side, we have 

`                  "database_name": {
                                Type:          schema.TypeString,
                                Optional:      true,
                                ConflictsWith: []string{"auth_database_name"},
                                Deprecated:    "use auth_database_name instead",
                        },
                        "auth_database_name": {
                                Type:          schema.TypeString,
                                Optional:      true,
                                ConflictsWith: []string{"database_name"},
                        },
`

However if we don't wan to break the backward compatibility in the client by now I could just add in the client a non exported json field , instead of modifying the interface signature and then not export the GetAuthDB, what do you think?
`
AuthDatabaseName string `json:"-"` //optional for auth
`

Reply:
Hi @gssbzn added the documentation line and below is the LDAP check